### ISSUE TITLE
[SUGGESTION] 7 Cool ideas to make this plugin the BEST report plugin!

### ISSUE THREAD

**Issue by User4716** [Syrent/VelocityReport#98]
### Server backend version

Paper 1.19.4

### Server Velocity version

Velocity

### VelocityReport version

3.4.0

### Describe the bug

I think there are not many report plugins out there that give you enough customizability and features or are even updated to the newest version of Minecraft. 
And even less or actually I think this is the only report plugin that has Velocity support. 
And that's why I am here to make the only good one that exists, better. 
If you would make a couple of features I suggest here, this would definitely be the go-to reporting plugin for not only Velocity networks, but any normal Paper server! And did I mention that the Book GUI is very intuitive and user friendly and also unique!😉 Yes, If you didn't notice. I love this plugin! (And even more, if you would do some of these features below ⬇️)
‎ 
‎ 
‎ 
### 🌟Here are some tips to make this plugin more advanced and better:
‎ 
‎ 
- ❌Ability to punish reporter for false report so then he can't make reports for another 3h, 1d, 7d, etc (Abusive report)
  Times are configurable in the configuration files. (Also you can specify custom commands to be run when punishing a false 
  reporter for example the first time. And a different command for the second time etc.)

![VelocityReport_3](https://user-images.githubusercontent.com/97033682/233791783-d11ea804-79b9-4c92-ac43-1bb0956505d6.png)

‎ 
- 📋Config file to modify every aspect of this plugin, for example, all text and maybe even all the commands to be run in different situations. And definitely all the report reasons too.

- ⏱️Reporting cooldown, for example 5m. Can be configured through config file and can be bypassed with a permission. There is by default 2 more tiers of reporting cooldown, so that if you have a higher permission, your report cooldown can be for example just 3 minutes.
`velocityreport.cooldown.default` = Default cooldown is 5m 
`velocityreport.cooldown.bypass` = Bypass cooldown
`velocityreport.cooldown.tier1` = Default tier 1 cooldown is 3m
`velocityreport.cooldown.tier2`= Default tier 2 cooldown is 2m
You can create infinite cooldown tiers in the config file.
‎ 
‎ 
‎ 
- ⚠️Admins will now get notified when a player gets reported (Requied permission: `velocityreport.admin.notify`) The notification text will say the reported player's name, the reporter's name, and the reason. 
Also, if you hover over the reported player's name it will say how many times the player has been reported 
in the **past hour, past week and all time.** There are a few other things, you can see them in the pictures. If you could modify everything about the messages and punishments it would be amazing!

![Admin notify](https://user-images.githubusercontent.com/97033682/233790581-c083e80c-aef1-43cb-9de7-a6c3f1481c0b.png)

![Admin notify 2](https://user-images.githubusercontent.com/97033682/233790604-a9c936a6-b3c7-4aff-9e83-f4f6d749d937.png)
‎ 

If you open the Moderation GUI, it would include some of the things you see below. ⬇️ This picture is from a plugin called TigerReports, this picture explains everything. I dont need to tell you what I want, (I was too lazy to type what I wanted so I just put that there) You can just look at that picture. You can implement as many features as you want from the picture below.
![TigerReports](https://user-images.githubusercontent.com/97033682/233791436-78b6efba-61e6-4bd9-bb92-e6cc822e5993.png)
‎ 
‎ 
‎ 
### **🛠️These things I want to the Moderation GUI:**
- See reported player and reporter.
- Report reason and severity 
- Approve report (Gives reward to a reporter if configured, by default it will give him report score. Punishes reported player, punishment commands configurable)
- Recjet report (Takes away report score from reporter) This should be used when the user's report is not 100% True.
- False report (Punishes reporter and takes away report points) This should be used when user is clearly spamming reports
- In-progress report (If you are not 100% sure, and don't want to reject yet)
- Helpers can vote for reports, they can't reject, or approve reports, they can vote for reports. They can react `⬆️Upvote` `⬇️Downvote` and `🤷Not sure` to reports, admins can also do this. This will significantly help admins resolve reports!
- Reporter's and Suspect's HP, Gamemode, Is online/offline, Location (XYZ), Server, World, UUID, IP-Adress, Alt accounts, Is on flying, active effects, Ability to teleport to both players and also the ability to teleport to the old location of both players when the report was made.
- And other things if you want. All of course is customizable, so that I can for example put my own placeholders. You can also specify a command to be run when teleporting to the player, such as /vanish
- When a report is rejected or approved, it will no longer be visible to normal users. You will need to have special permission to see approved and rejected reports. You can permanently remove reports too if you want with a permission.
- You can add notes to a report, so if you plan to set the state to "In progress" there will be notes for other admins to see.

‎ 
### **✨Other features:**
- Automatic command execution when a specific report amount is reached.

- Add a new placeholder %velocityreport_report_reason%
‎ ‎ 
- `/myreports` This command will open a Book GUI where you can see 2 options: "My reports" and "Reports to me"
  If you click "My reports" I can see all reports that I've made. If I click "Reports to me" I can see all the reports that I have gotten.
![VelocityReport_4](https://user-images.githubusercontent.com/97033682/233794305-08629af0-0247-475a-b3cb-782fadbdb3c6.png)

**My reports:** If you are an admin you will see a third option called "Manage reports" If you click it it will open an inventory GUI where you can manage all reports, more on that later.
![VelocityReport_5](https://user-images.githubusercontent.com/97033682/233796713-1b956d4a-d2ab-4a5d-ab04-0e7f8dcaeb7f.png)

**Reports to me:**
![VelocityReport_6](https://user-images.githubusercontent.com/97033682/233797226-f3c715b4-2d24-44f4-a872-3ee243708bae.png)

‎ 
- Add a new command that opens an Inventory GUI where you can manage all reports. If you are an admin you can also see a link to open this menu in the /myreports Book menu.
When you open this menu you have 5 options. `📋Latest reports` `🟢Resolved reports` `🔴Recjected reports` 
`🟡Reports in progress` `🪙Report point leaderboard` `👨‍🦱Most reported players` `🪓False reports`

![VelocityReport_7](https://user-images.githubusercontent.com/97033682/233798239-679567b4-0c60-4066-930d-01837333dc83.png)

**Latest reports:**
![VelocityReport_8_FANCY](https://user-images.githubusercontent.com/97033682/233799672-ae1900af-019f-4046-afbe-5528c0dfa81f.png)

So.. Thats all, it actually took mee longer than I thought to make this. It took 7 hours. Yes, idk why I used my whole day to this, but now at least you have ideas 😂

### Expected behavior

‎ 

### Steps to reproduce

‎ 

### Additional info

`‎v1.0`

### Checklist

- [X] I am running latest version of the plugin
- [X] I have read the wiki to make sure it's not an issue with configuration
- [ ] I ticked all of the boxes without actually reading them

**Comment by Syrent** [comment]
Thank you for your great suggestions, as always. I really appreciate the effort you put into making this detailed issue (I don't think I could have done it better myself 😅).

Regarding the "Reports that I've gotten" section, I believe it should not exist because reporters should have a certain level of privacy.

As for the notification feature in the chat for admins, I want to let you know that it already exists in the plugin, but I'll definitely work on improving it in future updates.

Lastly, I think the new menus and other features you suggested are fantastic and would greatly enhance the plugin's functionality. I'll do my best to add them as soon as possible.

Thank you again for your valuable input, it really helps me improve the plugin.

**Comment by User4716** [comment]
Yes, I didn't think you would actually consider making those features, but I really am thankful that you are ready to make some of those features. 😉

And the "Reports that I've gotten" section, is in my opinion good to have. 

When I made that I thought to myself that would it be a good idea to share the name that reported me. 
And I came to the conclusion that if it was not visible, it could cause report trolling. (Report trolling is making false reports and if there are no admins online, it would be more annoying to the falsely reported player if there is no name visible if there is multiple false reports.)

But now I think that the best solution is, that server owners can choose what to do.
They can configure it to show the names or hide the names in the config files. 👍
Or even better, when someone is making a report, they can choose to show/hide their name ✨

Thank you for reading, I actually have posted so many suggestions for you, so take your time on making these features, I think you want them to be polished and well made and not rushed. 😊
Hope you have the motivation to develop these ideas 
<3

**Comment by User4716** [comment]
Suggestion updated to 2.0. Mainly I clarified some things that might of been not well explained or not explained at all.
